### DISCUSSION TITLE
Optional Select

### DISCUSSION THREAD

**Discussion by Anon-136** [chakra-ui/zag#1708] [General]
https://github.com/user-attachments/assets/b6e4877d-e123-4872-8a09-9316b87d6294

I think it is normal to have an optional select field where you can select and unselect a value.

How would you recommend a way to make select unselectable in zagjs?

Should it be a new prop? How can I extend the select state machine?

or is it better to just use ClearTrigger?

**Comment by segunadebayo** [comment]
This is a good idea. We considered adding support for this in Select, Tabs, and other "selectable" components.

Feel free to pick this up if you want, otherwise, we might get to it in the future.

**Comment by Anon-136** [comment]
After some consideration, I think it's more clear to add clear icon button instead of unselect which are not the standard

![image](https://github.com/user-attachments/assets/f0ae146a-b01a-48ef-afb5-f7abae26688c)
